### ISSUE TITLE
describe.only() does not work when string is the same as another describe() block

### ISSUE THREAD

**Issue by c32hedge** [cypress-io/cypress#5345]
### Current behavior:

If you have multiple `describe` blocks with the same description string, marking one of them with `describe.only` will run all `describe` blocks with the same description instead of just the one with `.only`:

![image](https://user-images.githubusercontent.com/8810829/66594315-e86a8f80-eb66-11e9-8781-1db003aad501.png)

### Desired behavior:

Marking a `describe` block with `.only` should "only" run that `describe` block, not others with the same description.

### Steps to reproduce: (app code and test code)

Run the following spec:

```js
describe('outer describe', () => {
  describe('inner describe', () => {
    it('works', () => {});
  });

  describe.only('inner describe', () => {
    it('works again', () => {});
  });

  describe('unique name', () => {
    it('still works', () => {});
  });

  describe('inner describe', () => {
    it('works yet again', () => {});
  });
});
```
Instead of just running one test with the description "works again", three tests are executed, in all three `describe('inner describe'...)` blocks.

### Versions

Cypress 3.4.1
Ubuntu 18.04
Chrome 77

**Comment by c32hedge** [comment]
It's worth noting that Mocha itself used to have this issue (see https://github.com/mochajs/mocha/issues/1834, fixed by https://github.com/mochajs/mocha/pull/1807).

**Comment by jennifer-shehane** [comment]
Cypress uses a Mocha version 2.5.3, this is fixed in later versions. Going to close this as a duplicate of https://github.com/cypress-io/cypress/issues/2528 - which will be released in Cypress 4.0.0 release.

**Comment by c32hedge** [comment]
Wow, so Cypress is still on a version that's over three years old :scream:

**Comment by jennifer-shehane** [comment]
I've confirmed that this will be fixed in upcoming 4.0 release. 

<img width="461" alt="Screen Shot 2020-01-29 at 3 44 22 PM" src="https://user-images.githubusercontent.com/1271364/73343064-49e97200-42ae-11ea-8714-fbda96413b57.png">

**Comment by flotwig** [comment]
Released in [`4.0.0`](https://on.cypress.io/changelog#4-0-0).
